### PR TITLE
Add dict validation using a namespace

### PR DESCRIPTION
**Pull request by garethj2**
Added dict (record) validation to HNamespace.

@rracariu in particular please pay attention to how I'm skipping the third check if the tag value is null. Do you agree this is the correct approach?

### CODE REVIEW

**Comment by garethj2** [comment]
After discussing the use cases more with @rracariu I've changed the code so it can validate against a specific tag.

**Review by rracariu — APPROVED** [review]
LGTM

File: src/core/HNamespace.ts [hunk 1]
```diff
@@ -1380,4 +1381,78 @@ export class HNamespace {
...
+			if (!this.has(name)) {
+				throw new LocalizedError({
+					message: `Cannot find def '${name}'`,
+					lex: 'cannotFindDef',
+					args: { name },
+				})
+			}
+
+			// 2. Does the def have a kind?
+			const kind = this.defToKind(name)
```
[the remark this answers is not in the document]
Comment (garethj2) [reply src/core/HNamespace.ts]: If there's a mandatory tag called equip, this gets the intended kind for the def and makes sure it matches. For example, we don't want to have equip as a string.

File: src/core/HNamespace.ts [hunk 2]
```diff
@@ -1380,4 +1381,78 @@ export class HNamespace {
...
+					message: `Kind mismatch. '${name}' is ${value.getKind()} not ${kind}`,
+					lex: 'kindMismatch',
+					args: { name, kind, valueKind: value.getKind() },
+				})
+			}
+
+			// 4. Are all the mandatory tags present?
+			for (const { defName } of this.implementation(name)) {
+				if (!dict.has(defName)) {
+					throw new LocalizedError({
```
[the remark this answers is not in the document]
Comment (garethj2) [reply src/core/HNamespace.ts]: @rracariu I think you're reviewing the outdated code for some reason. Regardless, this logic is not too different in the updated files. 

The 'defName' exists as it's part of the implementation (or from the super types in the new code). The kind is validated accordingly.

**Pull request merged**